src/sql/workbench/services/notebook/browser/sql/sqlSessionManager.ts

```diff
@@ -133,6 +133,16 @@ export class SqlSessionManager implements nb.SessionManager {
 		}
 		return Promise.resolve();
 	}
+
+	shutdownAll(): Thenable<void> {
+		return Promise.all(SqlSessionManager._sessions.map(session => {
+			return this.shutdown(session.id);
+		})).then();
```

Comment:
Why the empty then() here?

Reply:
Also, do we need a catch() callback here to avoid unhandled rejections?

Reply:
FYI - the implementation here can return a promise (since they're inherently Thenables). So you can do that and then just make this async to make the code even cleaner

Reply:
Without the .then() I get an errr `Type 'void[]' is not assignable to type 'void'` since Promise.all returns an array of values. 

I don't think catch() is needed since shutdown() calls disconnect() which catches any error and logs it 
https://github.com/microsoft/azuredatastudio/blob/6b4c98892c1b2f9e8d71de58838af67e649ac109/src/sql/workbench/services/notebook/browser/sql/sqlSessionManager.ts#L427-L431